include/s3select_functions.h

```diff
@@ -1765,35 +1554,31 @@ bool operator()(bs_stmt_vec_t* args, variable* result) override
     iter++;
     base_statement* main_expr = *iter;
 
-    if (constant_state == false){
-      like_expr_val = like_expr->eval();
-      escape_expr_val = escape_expr->eval();
-
-      if (like_expr_val.type != value::value_En_t::STRING)  {
-        throw base_s3select_exception("like expression must be string");
-      }
-
-      if (escape_expr_val.type != value::value_En_t::STRING)  {
-        throw base_s3select_exception("esacpe expression must be string");
-      }
+    if (constant_state == false)
+    {
+      param_validation(escape_expr, like_expr);
 
       std::vector<char> like_as_regex = transform(like_expr_val.str(), *escape_expr_val.str());
       std::string like_as_regex_str(like_as_regex.begin(), like_as_regex.end());
-      compiled_regex = std::regex(like_as_regex_str);
+      compiled_regex.reset(new RE2(like_as_regex_str));
```

Comment:
please use std::make_unique (instead of new)

Reply:
Done